src/library/features/crates/cratefeature.cpp

```diff
@@ -154,69 +157,110 @@ bool CrateFeature::dropAcceptChild(const QModelIndex& index, QList<QUrl> urls,
         // Adds track, does not insert duplicates, handles unremoving logic.
         trackIds = m_pTrackCollection->getTrackDAO().addMultipleTracks(files, true);
     }
-    qDebug() << "CrateFeature::dropAcceptChild adding tracks"
-            << trackIds.size() << " to crate "<< crateId;
+    //qDebug() << "CrateFeature::dropAcceptChild adding tracks"
+    //        << trackIds.size() << " to crate "<< crateId;
     // remove tracks that could not be added
     for (int trackIdIndex = 0; trackIdIndex < trackIds.size(); ++trackIdIndex) {
         if (!trackIds.at(trackIdIndex).isValid()) {
             trackIds.removeAt(trackIdIndex--);
         }
     }
+    
+    // Request a name for the crate if it's a new crate
+    if (crateId < 0) {
+        QString name = getValidCrateName();
+        if (name.isNull()) {
+            return false;
+        }
+        
+        crateId = m_crateDao.createCrate(name);
+        // An error happened
+        if (crateId < 0) {
+            return false;
+        }
+    }
+    
     m_crateDao.addTracksToCrate(crateId, &trackIds);
     return true;
 }
 
 bool CrateFeature::dragMoveAcceptChild(const QModelIndex& index, QUrl url) {
     int crateId = crateIdFromIndex(index);
-    if (crateId == -1) {
-        return false;
-    }
+    // (timrae) why don't need to check for -1?
     bool locked = m_crateDao.isCrateLocked(crateId);
     bool formatSupported = SoundSourceProxy::isUrlSupported(url) ||
             Parser::isPlaylistFilenameSupported(url.toLocalFile());
     return !locked && formatSupported;
 }
 
-void CrateFeature::bindWidget(WLibrary* libraryWidget,
-                              KeyboardEventFilter* keyboard) {
-    Q_UNUSED(keyboard);
-    WLibraryTextBrowser* edit = new WLibraryTextBrowser(libraryWidget);
-    edit->setHtml(getRootViewHtml());
-    edit->setOpenLinks(false);
-    connect(edit, SIGNAL(anchorClicked(const QUrl)),
+QWidget* CrateFeature::createPaneWidget(KeyboardEventFilter *pKeyboard, 
```

Comment:
When the create pane widget is called always must return a widget without parent, then the in the `LibraryPaneManager` a parent is assigned. Maybe I'll add a coment.

See [src/library/librarypanemanager.cpp:40](https://github.com/daschuer/mixxx/blob/jmigual-library-redesign/src/library/librarypanemanager.cpp#L40)

```c++
for (LibraryFeature* f : m_features) {
    //f->bindPaneWidget(pPaneWidget, pKeyboard, m_paneId);
    
    QWidget* pFeaturePaneWidget = f->createPaneWidget(pKeyboard, m_paneId);
    if (pFeaturePaneWidget == nullptr) {
        continue;
    }
    pFeaturePaneWidget->setParent(pPaneWidget);
    pPaneWidget->registerView(f, pFeaturePaneWidget);
}
```